pandas/tests/arrays/test_datetimelike.py

```diff
@@ -774,8 +795,25 @@ def test_casting_nat_setitem_array(array, casting_nats):
     ids=lambda x: type(x).__name__,
 )
 def test_invalid_nat_setitem_array(array, non_casting_nats):
+    msg = (
+        r"'value' should be a 'Timestamp', 'NaT', or array of those. "
+        r"Got 'timedelta64' instead.|"
+        r"'value' should be a 'Timedelta', 'NaT', or array of those. "
+        r"Got 'datetime64' instead.|"
+        r"'value' should be a 'Timedelta', 'NaT', or array of those. "
+        r"Got 'int' instead.|"
+        r"'value' should be a 'Timestamp', 'NaT', or array of those. "
+        r"Got 'int' instead.|"
+        r"'value' should be a 'Period', 'NaT', or array of those. "
+        r"Got 'datetime64' instead.|"
+        r"'value' should be a 'Period', 'NaT', or array of those. "
+        r"Got 'timedelta64' instead.|"
+        r"'value' should be a 'Period', 'NaT', or array of those. "
+        r"Got 'int' instead."
+    )
```

Comment:
We are trying to use the pattern of:

```python
msg = "|".join(
    [
        "foo",
        "bar,
        "baz"
    ]
)

```

Reply:
After having a second look, I *think*  we can have this as a single string

```suggestion
    msg = (
        "'value' should be a '(Timestamp|Timedelta|Period)', 'NaT', or array of those. "
        "Got '(timedelta64|datetime64|int)' instead."
    )
```

Reply:
assertions fail when I use this,

```
E               AssertionError: Pattern "'value' should be a '(Timestamp|Timedelta|Period)','NaT', or array of those. Got '(timedelta64|datetime64|int)' instead." not found in "'value' should be a 'Timedelta', 'NaT', or array of those. Got 'datetime64' instead."
```



I'm trying to investigate as follows,

```python
>>> import re
>>> msg = "'value' should be a 'Timedelta', 'NaT', or array of those. Got 'datetime64' instead."
>>> pat = r"'value' should be a '(Timestamp|Timedelta|Period)', 'NaT', or array of those. Got '(timedelta64|datetimee64|int)' instead."
>>> re.search(pat, msg)
>>>
```

Reply:
it looks like missing a space after comma

Reply:
Ah my bad.